features/telemetry/common/src/main/java/org/opennms/netmgt/telemetry/common/ipc/TelemetrySinkModule.java

```diff
@@ -171,6 +171,10 @@ public boolean isBlockWhenFull() {
 
     @Override
     public Optional<String> getRoutingKey(final TelemetryProtos.TelemetryMessageLog message) {
+        // Allow the queue configuration to drive whether or not the routing key is used. Default to true.
+        if (!queueConfig.getUseRoutingKey().orElse(true)) {
+            return Optional.empty();
+        }
         return Optional.of(String.format("%s@%s:%d", message.getLocation(), message.getSourceAddress(), message.getSourcePort()));
```

Comment:
I had to read it three times to get the negation. Maybe this is simpler to read?
```suggestion
        if (queueConfig.getUseRoutingKey().orElse(true)) {
            return Optional.of(String.format("%s@%s:%d", message.getLocation(), message.getSourceAddress(), message.getSourcePort()));
        } else {
            return Optional.empty();
        }
```

Reply:
Ack.